app/models/services/service_key.rb

```diff
@@ -0,0 +1,95 @@
+module VCAP::CloudController
+  class ServiceKey < Sequel::Model
+    class InvalidAppAndServiceRelation < StandardError; end
+
+    many_to_one :service_instance
+
+    export_attributes :name, :service_instance_guid, :credentials, :syslog_drain_url
+
+    import_attributes :name, :service_instance_guid, :credentials, :syslog_drain_url
+
+    alias_attribute :broker_provided_id, :gateway_name
+
+    delegate :client, :service, :service_plan, to: :service_instance
+
+    plugin :after_initialize
+
+    encrypt :credentials, salt: :salt
+
+    def to_hash(opts={})
+      if !VCAP::CloudController::SecurityContext.admin? && !service_instance.space.developers.include?(VCAP::CloudController::SecurityContext.current_user)
+        opts.merge!({ redact: ['credentials'] })
+      end
+      super(opts)
+    end
+
+    def in_suspended_org?
+      space.in_suspended_org?
+    end
+
+    def space
+      service_instance.space
+    end
+
+    def validate
+      validates_presence :name
```

Comment:
yes, should be `validate_presence` not `validate_db_presence` in the test case, have modified and resubmitted the PR